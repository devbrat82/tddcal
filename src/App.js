import React, { useState } from 'react';
import './App.css';
import add from './utile/add';

function App() {
  const [input, setInput] = useState('');
  const [result, setResult] = useState(null);
  const [error, setError] = useState('');

  const handleCalculate = () => {
    try {
      setError('');
      const sum = add(input);
      setResult(sum);
    } catch (err) {
      setError(err.message);
    }
  };
  return (
    <div align="center">
    <h2>String Calculator</h2>
    <input
      type="text"
      value={input}
      onChange={(e) => setInput(e.target.value)}
      placeholder="Enter numbers"
    />
    <button onClick={handleCalculate}>Calculate</button>
    {result !== null && <p>Result: {result}</p>}
    {error && <p style={{ color: 'red' }}>{error}</p>}
  </div>
  );
}

export default App;
