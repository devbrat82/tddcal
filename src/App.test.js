import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

describe('String Calculator', () => {
test('renders String Calculator link', () => {
  render(<App />);
  const linkElement = screen.getByText(/String Calculator/i);
  expect(linkElement).toBeInTheDocument();
});

test('matches the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders the input and button correctly', () => {
  render(<App />);
  expect(screen.getByPlaceholderText('Enter numbers')).toBeInTheDocument();
  expect(screen.getByText('Calculate')).toBeInTheDocument();
});

test('calculates the sum correctly for valid input', () => {
  render(<App />);
  
  fireEvent.change(screen.getByPlaceholderText('Enter numbers'), { target: { value: '1,2,3' } });
  fireEvent.click(screen.getByText('Calculate'));
  
  expect(screen.getByText('Result: 6')).toBeInTheDocument();
});

test('displays an error message for invalid input', () => {
  render(<App />);
  
  fireEvent.change(screen.getByPlaceholderText('Enter numbers'), { target: { value: '1,-2,3' } });
  fireEvent.click(screen.getByText('Calculate'));
  
  expect(screen.getByText('negative numbers not allowed: -2')).toBeInTheDocument();
  expect(screen.queryByText('Result:')).toBeNull();
});
});
