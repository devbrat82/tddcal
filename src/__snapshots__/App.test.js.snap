// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String Calculator matches the snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
  >
    <h2>
      String Calculator
    </h2>
    <input
      placeholder="Enter numbers"
      type="text"
      value=""
    />
    <button>
      Calculate
    </button>
  </div>
</DocumentFragment>
`;
